sdk/dotnet/Pulumi/Serialization/Converter.cs

```diff
@@ -476,8 +476,8 @@ static bool CheckEnumType(Type targetType, Type underlyingType)
             }
         }
 
-        private static ConstructorInfo GetPropertyConstructor(Type outputTypeArg)
+        private static ConstructorInfo? GetPropertyConstructor(Type outputTypeArg)
             => outputTypeArg.GetConstructors(BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance).FirstOrDefault(
-                c => c.GetCustomAttributes<OutputConstructorAttribute>() != null);
+                c => c.GetCustomAttribute<OutputConstructorAttribute>() != null);
```

Comment:
This also looked like a bug as [`System.Attribute.GetCustomAttributes`](https://docs.microsoft.com/en-us/dotnet/api/system.attribute.getcustomattributes) never returns `null` so this filter is redundant and this method would always return all ctors:
> Returns ... An Attribute array that contains the custom attributes applied to element, _or an empty array if no such custom attributes exist_.